### PR TITLE
Fix torch.where signature mismatch that was caused by torchgen

### LINKED ISSUE #91003: torch.where signature mismatch
### 📚 torch.where signature mismatch

Based on the page of `torch.where` [1], the signature of that function should be
`torch.where(condition, x, y)`

When i try to call such function with:
`torch.where(condition=t, x=1, y=2)`
where `t` is a boolean `torch.tensor` i get the following error:

```
TypeError: where() received an invalid combination of arguments - got (Tensor, x=int, y=int), but expected one of:
 * (Tensor condition)
 * (Tensor condition, Tensor input, Tensor other)
      didn't match because some of the keywords were incorrect: x, y
 * (Tensor condition, Number self, Tensor other)
      didn't match because some of the keywords were incorrect: x, y
 * (Tensor condition, Tensor input, Number other)
      didn't match because some of the keywords were incorrect: x, y
 * (Tensor condition, Number self, Number other)
      didn't match because some of the keywords were incorrect: x, y
```

This means that the keywords of the function are: `condition, input, other` `instead of condition, x, y`, as can be seen in the source code [2].

[1]: https://pytorch.org/docs/stable/generated/torch.where.html
[2]: https://github.com/pytorch/pytorch/blob/583d216c1a767fcb9a8fc33fc2f7a19ff20eba95/aten/src/ATen/native/TensorCompare.cpp#L511

### Suggest a potential alternative/fix

Is it possible to change the signature of the where function by changing the keyword names. I would avoid changing the documentation and leaving such naming as most of the frameworks use `where(condition, x, y)`, thus PyTorch should also adopt that signature.
I would be honored to do such contribution.

cc @albanD

### PR DESCRIPTION
**Pull request by LucaLumetti**
Fixes #91003 

cc @ezyang @gchanan

### CODE REVIEW

**Review by albanD — APPROVED** [review]
Awesome! Thanks

File: torch/_torch_docs.py [hunk 1]
```diff
@@ -12490,27 +12490,27 @@ def merge_dicts(*dicts):
 add_docstr(
     torch.where,
     r"""
-where(condition, x, y, *, out=None) -> Tensor
+where(condition, input, output, *, out=None) -> Tensor
```
Comment (albanD) [inline torch/_torch_docs.py:12493]: wait this should be input and other not output right?

**Review by albanD — CHANGES_REQUESTED** [review]
Actually name is not correct?

File: torch/_torch_docs.py (hunk 1 above)
Comment (LucaLumetti) [reply torch/_torch_docs.py]: Yes, you are right, I've messed up for some unknown reason sorry!
I fixed it in the last commit

**Review by albanD — APPROVED** [review]
Thanks!

**Pull request closed without merging**